adoc/chapters/opencl_backend.adoc

```diff
@@ -149,15 +149,17 @@ combination of the two.
 
 The SYCL system presents the user with a set of devices, grouped into some
 number of platforms.
-The device version is an indication of the device's
-capabilities, as represented by the device information returned by the
-[code]#sycl::device::get_info()# member function.  Examples of attributes
-associated with the device version are resource limits and information
-about functionality beyond the requirements in the <<core-spec>>.
-The version returned corresponds to the highest version of the OpenCL
-specification for which the device is conformant, but is not higher than
-the version of the device's platform which bounds the overall capabilities
-of the runtime operating the device.
+
+The version returned by [code]#sycl::platform::get_info()# and
+[code]#sycl::device::get_info()# for the attributes
+[code]#info::platform::driver_version# and [code]#info::device::driver_version#
+respectively, represents the OpenCL software driver version in the form:
+[code]#major_number.minor_number#.
```

Comment:
Is it the intent that this number corresponds to the OpenCL *specification* version?  For example, "1.2" would indicate that the backend conforms to OpenCL 1.2 and "3.0" would indicate that it conforms to OpenCL 3.0?

When I read this before, I though the intent was for each vendor to return some vendor-specific version number of the device driver.  This is what DPC++ does currently.  For example, we currently return this for one of our OpenCL devices:

`"2021.13.11.0.23_160000"`

Reply:
I've re-worked this paragraph into a table referring to the OpenCL properties which I think should be much clearer.

However I noticed that in the comments this section was referring to another one from the main spec: [Platform mixed version support](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_platform_mixed_version_support).

But I can't really make sense of what this section is trying to say or how this part of the OpenCL backend spec was addressing it, does it have something to do with the abandoned idea of mixing SYCL 1.2 and 2.2 devices as @keryell suggested?

Reply:
Yes, I think this is to suggest that while the top-level supported version has some SYCL versions but using some platform might supporting only an older SYCL version. But since we no longer have true feature inclusion, I am not sure it still makes sense.